src/storage/mod.rs

```diff
@@ -1428,17 +1278,58 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
     /// return its MVCC info.
     pub fn mvcc_by_start_ts(
         &self,
-        ctx: Context,
-        start_ts: TimeStamp,
+        cmd: Command,
         callback: Callback<Option<(Key, MvccInfo)>>,
     ) -> Result<()> {
-        let cmd = commands::MvccByStartTs::new(start_ts, ctx);
         self.schedule(cmd, StorageCallback::MvccInfoByStartTs(callback))?;
         KV_COMMAND_COUNTER_VEC_STATIC.start_ts_mvcc.inc();
         Ok(())
     }
 }
 
+/// Get a single value.
+pub struct PointGetCommand {
+    pub ctx: Context,
+    pub key: Key,
+    /// None if this is a raw get, Some if this is a transactional get.
+    pub ts: Option<TimeStamp>,
+}
+
+impl PointGetCommand {
+    pub fn from_get(request: &mut GetRequest) -> Self {
+        PointGetCommand {
+            ctx: request.take_context(),
+            key: Key::from_raw(request.get_key()),
+            ts: Some(request.get_version().into()),
+        }
+    }
+
+    pub fn from_raw_get(request: &mut RawGetRequest) -> Self {
+        PointGetCommand {
+            ctx: request.take_context(),
+            key: Key::from_raw(request.get_key()),
```

Comment:
@breeswish So for rawKV we will encode the key? That's weird.

Reply:
we shouldn't

Reply:
This code was just moved, not changed. The current code also does this: https://github.com/tikv/tikv/blob/0c9669dd1727b4142d9a5fe248d99a7752e81d17/src/storage/txn/commands.rs#L32.

If this should be changed, then it should be done in a separate PR.

Reply:
I think maybe you misunderstood. `Key::from_raw` is actually encoding a key. For raw kv operations, key should not be encoded and should be written to or read from the engine as-is.

Reply:
Oh I got it. the original code is already like this. It should be incorrect.

Reply:
This problem is introcuded by https://github.com/tikv/tikv/pull/5598 . The unit test uses a wrong way [here](https://github.com/tikv/tikv/pull/5598/files#diff-a2828e4f71806f3e4623d048057803d4R3391) to create the command, so the test didn't fail.
@zhangjinpeng1987 PTAL.